packages/common/src/utils/NetworkUtil.ts

```diff
@@ -17,5 +17,51 @@ export const NetworkUtil = {
 
   getFirstNetworkByNamespace(networks: CaipNetwork[] | undefined, namespace: ChainNamespace) {
     return this.getNetworksByNamespace(networks, namespace)[0]
+  },
+
+  getNetworkNameByCaipNetworkId(caipNetworkId: CaipNetworkId): string | undefined {
```

Comment:
@enesozturk can't we make it so this receives a network list and we just feed it in from where we call it ?

Reply:
Totally right, we can read the requested networks and check the name from there

Reply:
@tomiir Can you review agian? 🙏🏽